test/unit/org/apache/cassandra/cql3/validation/operations/DeleteTest.java

```diff
@@ -162,6 +164,39 @@ public void testDeletion() throws Throwable
                    row("abc", 4, "xyz", "some other value"));
     }
 
+    @Test
+    public void testDeletionWithContains() throws Throwable
+    {
+        createTable("CREATE TABLE %s (a int, b frozen<map<int, int>>, c int, primary key (a, b))");
+
+        Map<Integer, Integer> testMap = new HashMap<>();
+        testMap.put(1, 1);
+        testMap.put(2, 2);
+        execute("INSERT INTO %s (a, b, c) VALUES (?, ?, ?)", 1, testMap, 3);
```

Comment:
We don't strictly need to insert data to verify that the `DELETE` query fails its validation. If we do so, it can be useful to verify that the data is still there after rejecting the `DELETE` query, so we verify that the `DELETE` query has been actually rejected. So, simplifying the code a little bit:
```java
createTable("CREATE TABLE %s (a int, b frozen<map<int, int>>, c int, PRIMARY KEY (a, b))");

Object[] row = row(1, map(1, 1, 2, 3), 3);
execute("INSERT INTO %s (a, b, c) VALUES (?, ?, ?)", row);
assertRows(execute("SELECT * FROM %s"), row);

assertInvalidMessage("Cannot use DELETE with CONTAINS",
                     "DELETE FROM %s WHERE a=1 AND b CONTAINS 1");

// verify that the previous invalid DELETE query hasn't been applied
assertRows(execute("SELECT * FROM %s"), row);
```
This can also be applied to `testDeletionWithContainsKey` and the new analogous tests on `UpdateTest`, wdyt?

Reply:
Agree, will update to also check that data is still there after delete.